apm-agent-core/src/main/java/co/elastic/apm/agent/report/IntakeV2ReportingEventHandler.java

```diff
@@ -37,22 +39,28 @@
 
     public static final String INTAKE_V2_URL = "/intake/v2/events";
     private final ProcessorEventHandler processorEventHandler;
-    private final Timer timeoutTimer;
+    private final ScheduledExecutorService timeoutTimer;
     @Nullable
-    private ApmServerReporter reporter;
-    @Nullable
-    private TimerTask timeoutTask;
+    private Runnable timeoutTask;
+    private final AtomicLong processed = new AtomicLong();
+
+    public IntakeV2ReportingEventHandler(ReporterConfiguration reporterConfiguration, ProcessorEventHandler processorEventHandler,
```

Comment:
```suggestion
   IntakeV2ReportingEventHandler(ReporterConfiguration reporterConfiguration, ProcessorEventHandler processorEventHandler,
```

Maybe also javadoc that this is intended for tests

Reply:
I removed the Clock abstraction (see below).